hal/src/thumbv6m/sercom/v1/i2c.rs

```diff
@@ -158,6 +184,12 @@ where
         if status.rxnack().bit_is_set() {
             return Err(I2CError::Nack);
         }
+
+        #[cfg(feature = "samd20")]
+        if status.lowtout().bit_is_set() {
+            return Err(I2CError::Timeout);
+        }
+        #[cfg(not(feature = "samd20"))]
         if status.lowtout().bit_is_set() || status.sexttout().bit_is_set()
             || status.mexttout().bit_is_set()
         {
```

Comment:
You might want to break this up into separate SAMD20/non-SAMD20 functions to avoid so many `#[cfg]`s, or at least into config blocks.